src/sentry/api/serializers/rest_framework/rule.py

```diff
@@ -103,22 +102,19 @@ def validate(self, attrs):
                 )
 
         # ensure that if a user has alert-filters enabled, they do not use old conditions
-        project = self.context["project"]
         conditions = attrs.get("conditions", tuple())
-        project_has_filters = features.has("projects:alert-filters", project)
-        if project_has_filters:
-            old_conditions = [
-                condition for condition in conditions if condition["id"] in MIGRATED_CONDITIONS
-            ]
-            if old_conditions:
-                raise serializers.ValidationError(
-                    {
-                        "conditions": "Conditions evaluating an event attribute, tag, or level are outdated please use an appropriate filter instead."
-                    }
-                )
+        old_conditions = [
+            condition for condition in conditions if condition["id"] in MIGRATED_CONDITIONS
```

Comment:
I think you might be able to remove this logic and `MIGRATED_CONDITIONS` entirely since there was a migration to move the data over - worth double checking but I don't think there should be any old conditions in the db and I don't think it's currently possible to get into this state (unless someone passed in a bad value via the API?)

Reply:
Yeah, this would need to be a bad condition passed in through the API. Do you think enough time has passed that we can remove the check?

Reply:
If 4 years isn't enough time idk how much time would be 😅

Reply:
If you want to be really conservative you can add logging if we do hit this line though

Reply:
I checked today—there are still a lot of rules with the old conditions 😬 so we can't remove the old conditions entirely, which means that we can't move MIGRATED_CONDITIONS.